reframe/core/schedulers/slurm.py

```diff
@@ -157,6 +157,8 @@ def _count_compatible_nodes(self, nodes):
         constraints = set()
         partitions = (set(self.sched_partition.split())
                       if self.sched_partition else set())
+        excluded_nodes = (set(self.sched_exclude_nodelist.split())
```

Comment:
As discussed offline, the `split()` here is restrictive. The `sched_exclude_nodelist` can get any value that Slurm accepts.

Reply:
Also, for debugging reasons, please implement the `__str__()` method of the `SlurmNode`. It could simply return the node name.